dart/lib/src/sentry_client.dart

```diff
@@ -108,7 +108,8 @@ class SentryClient {
       return _sentryId;
     }
 
-    if (_options.platformChecker.platform.isAndroid &&
+    if (!_options.platformChecker.isWeb &&
+        _options.platformChecker.platform.isAndroid &&
```

Comment:
Can you check if this PR https://github.com/getsentry/sentry-dart/pull/1366/ would fix the issue as well?

Reply:
The PR still checks `isAndroid` and `isWeb` only but I don't see how `isWeb` can be true but also `isAndroid` at the same time.
@denrase have you tested on Web?

Reply:
> I don't see how isWeb can be true but also isAndroid at the same time.

It certainly was, although I don't know the details.

Reply:
@mnordine were you running a Flutter web app on a Mobile Android browser by any chance?

Reply:
Nope, nothing to do with Flutter.

Reply:
So were you running a pure Dart Web app?

Reply:
It's right in the issue: https://github.com/getsentry/sentry-dart/issues/979

Dart web app, affects all browsers on Android.

Reply:
So still makes sense, @mnordine can you merge the main branch into your branch and fix the merge conflict? that part you changed moved to a different method.
Also please add a test :)
Thank you!

Reply:
> Can you check if this PR https://github.com/getsentry/sentry-dart/pull/1366 would fix the issue as well?

Looks like the bug still exists when run against `main`, after merging in the above PR.

> can you merge the main branch into your branch and fix the merge conflict?

I don't think I'll have a chance today, but will get to it when I can.

Reply:
> Also please add a test :)

I added a test, but I was kind of taking a shot in the dark, not being familiar with your test suite, practices, etc...